linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -354,6 +342,30 @@ where
         }
         Ok(ChainInfoResponse::new(info, self.0.config.key_pair()))
     }
+
+    fn check_blob_size(
+        content: &BlobContent,
+        policy: &ResourceControlPolicy,
+    ) -> Result<(), WorkerError> {
+        let blob_size = match content {
+            BlobContent::Data(bytes) => bytes.len(),
+            BlobContent::ContractBytecode(compressed_bytecode)
+            | BlobContent::ServiceBytecode(compressed_bytecode) => {
+                ensure!(
+                    compressed_bytecode.decompressed_size_at_most(policy.maximum_bytecode_size)?,
+                    WorkerError::BytecodeTooLarge
+                );
+                compressed_bytecode.compressed_bytes.len()
+            }
+        };
```

Comment:
How about a new method `fn size(&self)` on `BlobContent`?

Reply:
Ah, I can see it's not that simple since for `ContractBytecode` and `ServiceBytecode` we do two checks 🤔

Reply:
Maybe; that would only save one line this function, though: We'd still need the `match`, to check the _decompressed bytecode_ size.

Reply:
And we should _not_ have a function that returns the bytecode size: We only unzip it up to the point where it exceeds the limit.